src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -0,0 +1,174 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import scala.Tuple2;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.ContigsCollection.loadContigsCollectionKeyedByAssemblyId;
+
+/**
+ * Implements a parser for parsing alignments of locally assembled contigs, and
+ * generating {@link ChimericAlignment}'s from the alignments, if possible.
+ */
+class AssemblyAlignmentParser implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    static final int ARTIFICIAL_CIGAR_MISMATCH = -1;
```

Comment:
Will go with the special value here as well, just like the case in `AlignmentRegion`